lib/src/sdk.ts

```diff
@@ -450,3 +475,78 @@ function getSwapStep(error: Error): string {
 
   return "UNKNOWN_STEP";
 }
+
+async function sellPayloadRequest({
+  fromAmount,
+  info,
+  account,
+  deviceTransactionId,
+  initialAtomicAmount,
+  providerId,
+  handleError,
+  getSellPayload,
+}: {
+  fromAmount: BigNumber;
+  info: SellInfo;
+  account: Account;
+  deviceTransactionId: string;
+  initialAtomicAmount: BigNumber;
+  providerId: string;
+  handleError: (error: Error) => void;
+  getSellPayload?: GetSellPayload;
+}) {
+  let recipientAddress, binaryPayload, signature, beData;
+  let amount = fromAmount;
+
+  // For providers that send us getSellPayload (Coinify)
+  if (getSellPayload !== undefined) {
+    const payloadRequest = getSellPayload;
+
+    const data = await payloadRequest(
+      deviceTransactionId,
+      account.address,
+      initialAtomicAmount
+    ).catch((error: Error) => {
+      const err = new PayloadStepError(error);
+      handleError(err);
+
+      throw error;
+    });
+
+    recipientAddress = data.recipientAddress;
+    amount = data.amount;
+    binaryPayload = data.binaryPayload;
+    signature = data.signature;
+    beData = data.beData;
+  } else {
+    // For all other providers
+    const payloadRequest = retriveSellPayload;
+
+    const data = await payloadRequest({
+      quoteId: info.quoteId!,
```

Comment:
Because typescript does not know how to infer the type here based on  `if (getSellPayload !== undefined){} else {}`
The alternative is to cast it as a string.

Reply:
maybe we could mark `quotoId` as required in the [SellInfo interface](https://github.com/LedgerHQ/exchange-sdk/pull/76/files#diff-e78c98103b01832d6cf8f383cdde2297dde41606e1b05d445df716e4a86d3bf5R81) ? if not, we could add some validation here. If `quoteId` is not defined then throw an error